### ISSUE TITLE
Proper .gitignore support

### ISSUE THREAD

**Issue by feross** [feross/standard#125]
This looks like it actually does the trick: https://www.npmjs.com/package/ignore

**Comment by tunnckoCore** [comment]
Why not just [parse gitignore](https://github.com/jonschlinkert/parse-gitignore) from cwd and then append defaults? And then update them like in the last [lines of the index.js](https://github.com/feross/standard/blob/master/index.js#L148-L152) which is weird btw.

**Comment by Flet** [comment]
Part of the issue is the gitignore patterns are not the same as monarch/glob patterns. The ignore package above looks looks like it reconciles these patterns.

**Comment by feross** [comment]
PR welcome! The `ignore` package looks like the way to go.

**Comment by jprichardson** [comment]
If we do this, could it be done in a 4.0 branch as couldn't it be stated that this could violate semver if someone needed to modify their `.gitignore` file to achieve the same results?

**Comment by jden** [comment]
it should follow the npm ignore algorithm: https://docs.npmjs.com/misc/developers#keeping-files-out-of-your-package

**Comment by feross** [comment]
Okay, I have this working. Should it be 3.x or 4.0.0?

@jprichardson The "ignore" property of package.json isn't going away. The .gitignore is purely additive; it's adding new ignored files. That makes me think this should be 3.x. No one's tests should break because of this. When I originally added .gitignore support, it was also done in a minor release.

Anyone have any thoughts (in the next 30 minutes) before I cut this release?

**Comment by feross** [comment]
Well, actually, I have another definitely breaking change (operators go before line break ) I wanted to get into 4.0.0. Might as well just release this change together with that one to be on the safe side.

**Comment by feross** [comment]
Okay, standard 4.0.0 is released! Enjoy :)

**Comment by dcousens** [comment]
> operators go before line break 

Excellent :)

**Comment by jden** [comment]
for the sake of anyone searching back through issues trying to figure out a change in ignore behavior like I was:

in ~3.0.0, package.json `"ignore": ["artifacts/"]` worked. with this change, in ~4.0.0, it must be `"ignore": ["artifacts/**"]`

**Comment by feross** [comment]
@jden That's odd. `"ignore": ["artifacts/"]` works just fine for me. What platform are you on? Node version? Standard version?

**Comment by jden** [comment]
@feross on `standard@4.0.1`

```sh
ci *= ~/dev/agilemd/api
 2:21> node -p "require('os').platform() + ' ' + require('os').arch()"
win32 x64

ci *= ~/dev/agilemd/api
 2:21> node -v
v0.10.36
```

**Comment by feross** [comment]
@jden I think I know what's causing this, but it's not easy for me to test on Windows.

Can you try using the code on the [`windows-gitignore-fix` branch](https://github.com/feross/standard/tree/windows-gitignore-fix) and see if that fixes the problem for you?

**Comment by jden** [comment]
@feross I tried that out and get the same behavior. Note, I'm setting my ignores in the `"standard"` property of the package.json, not in .gitignore.

**Comment by feross** [comment]
@jden Yep, I'm aware you're using the standard.ignore property of package.json :) Sorry, I made a [silly mistake](https://github.com/feross/standard/commit/2e5e97a324eb124bc80fdc9af27331dcd4ddad5e). Can you try that branch once more?

**Comment by jden** [comment]
@feross yep, confirmed working!

**Comment by feross** [comment]
@jden Cool, thanks for the help. I released this fix as 4.1.1. :+1:
